src/Dapr.Actors/Runtime/Actor.cs

```diff
@@ -267,24 +257,20 @@ protected Task UnregisterReminderAsync(string reminderName)
         /// Specify negative one (-1) milliseconds to disable periodic signaling.</param>
         /// <returns>Returns IActorTimer object.</returns>
         protected Task<IActorTimer> RegisterTimerAsync(
-            Func<object, Task> asyncCallback,
+            string callback,
             object state,
             TimeSpan dueTime,
             TimeSpan period)
         {
-            return this.RegisterTimerAsync(null, asyncCallback, state, dueTime, period);
+            return this.RegisterTimerAsync(null, callback, Encoding.UTF8.GetBytes(state.ToString()), dueTime, period);
```

Comment:
`state.ToString()`

Unfortunately, this isn't idiomatic in .NET. There are APIs that accept an object `state` parameter, but the expectation is that they will give you the literal same object when your timer fires.  This doesn't follow that pattern.

Think about about the code that you'd write - you'd pass in a `Widget` here, and then your callback would have to accept bytes, not a `Widget`.

My suggestion is to just remove it this overload, and leave the `byte[]` one. That makes it clear that your data needs to be serialized/deserialized.